eng/doc/fips/README.md

```diff
@@ -10,40 +10,87 @@ The Go `crypto` package is not FIPS certified, and the Go team has stated that i
 
 It is important to note that an application built with Microsoft's Go toolchain and running in FIPS compatible mode is not FIPS compliant _per-se_. It is on the application development team to use FIPS-compliant crypto primitives and workflows. The crypto runtime will fall back to Go standard library crypto in case it cannot provide a FIPS-compliant implementation, e.g. when hashing a message using `crypto/md5` hashes or when using an AES-GCM cipher with a non-standard nonce size.
 
-## Usage
+## Usage: Build
+
+### Go 1.18
+
+In Go 1.18 and earlier, the Microsoft Go FIPS-compatible builds are maintained in the `microsoft/dev.boringcrypto*` branches. Only a Linux implementation using OpenSSL is supported for these versions.
+
+1. Get a `1.18-fips` build of the Microsoft Go toolset. See [the microsoft/go readme](https://github.com/microsoft/go#binary-distribution) for options.
+1. Build your Go program using this Go toolset.
+1. The built program now includes the ability to use OpenSSL crypto and FIPS compatibility mode at runtime.
+
+The whole OpenSSL functionality can be disabled even while using a `1.18-fips` Go toolset by building your program with `-tags gocrypt`.
+
+### Go 1.19+
+
+In Go 1.19 onward, the FIPS-related changes are maintained in the `microsoft/release-branch.go*` branches in this repository.
+
+1. Get the standard Microsoft build of Go. There is no separate `-fips` build. See [the microsoft/go readme](https://github.com/microsoft/go#binary-distribution) for options.
+1. Enable the desired GOEXPERIMENT and build your program.
+    * To build for Linux/OpenSSL, for example:
+      ```sh
+      GOEXPERIMENT=opensslcrypto go build ./myapp
+      ```
+    * To build for Windows/CNG in PowerShell, for example:
+      ```pwsh
+      $env:GOEXPERIMENT = "opensslcrypto"
+      go build ./myapp
+      ```
+1. The built program will use the specified platform-provided cryptographic library whenever it calls a Go standard library crypto API, and FIPS compatibility can be enabled at runtime.
+
+## Usage: Runtime
 
-FIPS compatibility mode, and therefore the OpenSSL crypto backend, can be enabled using any of these options:
+A program built with Go 1.19+ and `opensslcrypto` always uses the OpenSSL library present on the system for crypto APIs. Likewise for `cngcrypto` and CNG. If the platform's crypto library can't be found or loaded, the Go program panics during initialization.
 
-- Explicitly setting the environment variable `GOFIPS=1`.
-- Implicitly enabling it by booting the Linux Kernel in FIPS mode.
+In Go 1.18 and earlier, the program uses OpenSSL only if the program is running in FIPS mode.
+
+The following sections describe how to enable FIPS mode.
+
+### Linux FIPS mode (OpenSSL)
+
+To set FIPS mode on Linux, use one of the following options. The first match wins:
```

Comment:
I would also add the option that neither `GOFIPS` nor kernel `fips_enabled` are set, in which case the FIPS mode is whatever is configured via standard OpenSSL configuration. Might also link to https://www.openssl.org/docs/fips.html.

Reply:
I also added a note that `GOFIPS=0` means Go will attempt to disable FIPS (not just *allow* it to be disabled), in case it isn't clear enough from the bullet point.